### PR TITLE
feature/RE-54-ratings-implementieren

### PR DESCRIPTION
**Pull request by Manu100001**

### CODE REVIEW

File: backend/src/models/ratings.model.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { Ratings } from '@interfaces/ratings.interface';
+
+const ratingsSchema: Schema = new Schema({
+  customerId: {
+    type: String,
+    required: true,
+    unique: false,
+  },
+
+  priductId: {
```
Comment (Samay97) [inline backend/src/models/ratings.model.ts]: typo

File: backend/src/dtos/ratings.dto.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  public customerId: string;
+
+  @IsString()
+  public productId: string;
+
+  @IsNumber()
+  public rating: number;
+
+  @IsDate()
+  public timestamp: Date;
```
Comment (Samay97) [inline backend/src/dtos/ratings.dto.ts:16]: which date format is this?

File: backend/src/controllers/ratings.controller.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { NextFunction, Request, Response } from 'express';
+import { CreateRatingsDto } from '@dtos/ratings.dto';
+import { Ratings } from '@interfaces/ratings.interface';
+import ratingService from '@services/ratings.service';
+
+class RatingController {
```
Comment (Samay97) [inline backend/src/controllers/ratings.controller.ts:6]: Maybe add endpoint and controller function to get a rating for a product combined.
=> 3,4 Stars with 34 ratings

File: backend/src/dtos/ratings.dto.ts (hunk 2 above)
Comment (Manu100001) [reply backend/src/dtos/ratings.dto.ts:16]: I guess: 1972-03-28T23:11:00

File: backend/src/models/ratings.model.ts (hunk 1 above)
Comment (Manu100001) [reply backend/src/models/ratings.model.ts]: Done [f84a1d4]

File: backend/src/controllers/ratings.controller.ts (hunk 3 above)
Comment (Manu100001) [reply backend/src/controllers/ratings.controller.ts:6]: [f84a1d4]

File: backend/src/controllers/ratings.controller.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  public getRatingsByProductId = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const productId: string = req.params.id;
+      const findRatings: Ratings[] = await this.ratingService.getRatingsByProductId(productId);
+
+      let ratings = 0;
+      length = findRatings.length;
+      for (let i = 0; i < length; i = i + 1) {
```
Comment (Samay97) [inline backend/src/controllers/ratings.controller.ts]: Works, but is not the best way in js.
Javascript has many array functions that can help you. 
I think a reduce approach like this should be a "better" solution.

Even there is an error with ratingsSum, 0 will be retuned as default AVG.

only as a small improvement

````javascript
let ratingsSum = findRatings.reduce(
  (previousValue, currentValue) => previousValue + currentValue.rating,
  0
);

const ratingAVG = ratingsSum / findRatings.length || 0;

const data = [ratingAVG, findRatings.length];
```

Comment (Manu100001) [reply backend/src/controllers/ratings.controller.ts]: [3aeae49] done

**Review by Samay97 — APPROVED** [review]

**Pull request merged**